### PR TITLE
Brings Heretics Up To Date

### PR DESCRIPTION
**Pull request by eeSPee**
## About The Pull Request

Brings heretics code up to date with these major changes:
* Tears in reality no longer inflict brain damage when analyzed, instead just give a bad moodlet
* Revised reality tears generation
* Tears in reality appear after a time, and slowly disappear after 15 minutes
* Heretic effects (runes, tears) are invisible to silicons
* Names and descriptions have been revised, thanks to @EOBGames
* Heart now offers a pick between 3 targets
* Doubled the charges sacrifices offer
* Heretics marks area work differently; marks are applied by grasp, and triggered with the blade
* Ash grasp blinds instead of yeeting
* Rust spread datum is revised/optimized

Up to date with the PRs:
* https://github.com/tgstation/tgstation/commit/283d140ca08c1cdc2f6d9f46e9458c29e270b905#diff-f13197ae0925553052a7e70a8cd27e9a97baf2b54f00182a36a7b4c7a25cfbc3
* https://github.com/tgstation/tgstation/commit/3190cc36038874803bf198f06b2eb0f48bf2724a#diff-52a3ee8ccf1b6f2e59a60d8ff609db64bd4c65424592664c508d0f657bd0cae4
* https://github.com/tgstation/tgstation/commit/7c123a2fe27206f8af35b8455dab7e7ba24b7156#diff-0bcd93446e8e6938a1603b628e91bbf6be72214112c70198f95f48bae9b44a67
* https://github.com/tgstation/tgstation/pull/54372
* https://github.com/tgstation/tgstation/pull/54371
* https://github.com/tgstation/tgstation/pull/54967
* https://github.com/tgstation/tgstation/pull/54475
* https://github.com/tgstation/tgstation/pull/53414
* https://github.com/tgstation/tgstation/pull/53396
* https://github.com/tgstation/tgstation/pull/52680

## Why It's Good For The Game

Heretics is out of date and tg made many QoL changes that improves performance and how heretics work.

## Changelog
:cl: EdgeLordExe, EOBGames, ShivCalev, Rohesie, Tad Hardesty, ATHATH, and eeSPee
tweak: Tears in reality no longer inflict brain damage when analyzed, instead just give a bad moodlet
tweak: Revised reality tears generation
tweak: Tears in reality appear after a time, and slowly disappear after 15 minutes
tweak: Coders rule, jannies drool
balance: Heretic effects (runes, tears) are invisible to silicons
spellcheck: Names and descriptions have been revised, thanks to @EOBGames
balance:  Heart now offers a pick between 3 targets
balance: Doubled the charges sacrifices offer
balance: Heretics marks area work differently; marks are applied by grasp, and triggered with the blade
balance: Ash grasp blinds instead of knockback
tweak: Rust spread datum is revised/optimized
/:cl:

### CODE REVIEW

File: code/modules/antagonists/eldritch_cult/eldritch_effects.dm
Comment (PowerfulBacon) [inline code/modules/antagonists/eldritch_cult/eldritch_effects.dm]: this is the shit that makes it invisible to non-heretics I'm pretty sure

Comment (eeSPee) [reply code/modules/antagonists/eldritch_cult/eldritch_effects.dm]: I'm not sure why I commented it out. I think it failed my build or whatever.

**Comment by eeSPee** [comment]
Sorry for so many Commits; dream maker is acting up and won't let me compile.

File: code/modules/antagonists/eldritch_cult/eldritch_effects.dm
Comment (eeSPee) [reply code/modules/antagonists/eldritch_cult/eldritch_effects.dm]: Ah; it should work now properly.

**Comment by heepox** [comment]
so much yes.

**Comment by bloons3** [comment]
@eeSPee 

This doesn't prevent silicons from seeing heretic runes/pierced realities?

https://github.com/tgstation/tgstation/pull/54372

Missing this PR

**Comment by Fronsis** [comment]
Why are they invisible to sillicons?

**Comment by bloons3** [comment]
> Why are they invisible to sillicons?

https://github.com/tgstation/tgstation/pull/54372

Cult runes are obscured from silicons as well, so consistency

File: code/datums/mood_events/generic_negative_events.dm
Comment (park66665) [inline code/datums/mood_events/generic_negative_events.dm]:
```suggestion
```

We already have `/datum/mood_event/gates_of_mansus`. It is about twenty lines above.

File: code/modules/antagonists/eldritch_cult/eldritch_effects.dm
Comment (park66665) [inline code/modules/antagonists/eldritch_cult/eldritch_effects.dm:76]: This change is from tgstation/tgstation#54371. Better credit it.

File: code/modules/antagonists/eldritch_cult/eldritch_antag.dm
Comment (park66665) [inline code/modules/antagonists/eldritch_cult/eldritch_antag.dm]: `on_death` part is from tgstation/tgstation#54252

File: code/modules/antagonists/eldritch_cult/eldritch_effects.dm
Comment (park66665) [inline code/modules/antagonists/eldritch_cult/eldritch_effects.dm:116]: This set of changes are from tgstation/tgstation#54967. And better change the comment.

File: code/modules/antagonists/eldritch_cult/eldritch_effects.dm
Comment (park66665) [inline code/modules/antagonists/eldritch_cult/eldritch_effects.dm:211]: This set of changes also includes changes from tgstation/tgstation#54475.

File: code/modules/antagonists/eldritch_cult/eldritch_effects.dm
Comment (park66665) [inline code/modules/antagonists/eldritch_cult/eldritch_effects.dm:214]: Movement of `. = ..()` is from tgstation/tgstation#53414

File: code/modules/antagonists/eldritch_cult/eldritch_effects.dm
Comment (park66665) [inline code/modules/antagonists/eldritch_cult/eldritch_effects.dm]: You didn't properly port tgstation/tgstation#54967.

File: code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm
Comment (park66665) [inline code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm]: Those variables are not used in this PR. Are you going to use them in future PRs?

File: code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm
Comment (park66665) [inline code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm:148]: You should restore these lines. We don't have `GLOB.human_list`

File: code/modules/antagonists/eldritch_cult/knowledge/ash_lore.dm
Comment (park66665) [inline code/modules/antagonists/eldritch_cult/knowledge/ash_lore.dm:37]: This change is from tgstation/tgstation#53396

File: code/modules/antagonists/eldritch_cult/knowledge/rust_lore.dm
Comment (park66665) [inline code/modules/antagonists/eldritch_cult/knowledge/rust_lore.dm]: This set of changes is from tgstation/tgstation#52680

**Review by park66665 — CHANGES_REQUESTED** [review]

File: code/modules/antagonists/eldritch_cult/eldritch_antag.dm
Comment (eeSPee) [reply code/modules/antagonists/eldritch_cult/eldritch_antag.dm]: Thanks. I missed this one.

File: code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm
Comment (eeSPee) [reply code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm:148]: No need; now checks if istype isntead of ishuman, see the lines below.

Comment (park66665) [reply code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm:148]: Ohhhhh there it is. I missed that one.

**Comment by eeSPee** [comment]
Thanks! Fixed the code and cited the proper PRs.

File: code/modules/antagonists/eldritch_cult/eldritch_effects.dm
Comment (park66665) [inline code/modules/antagonists/eldritch_cult/eldritch_effects.dm]:
```suggestion
	if(istype(human_user) && !IS_HERETIC(human_user) && !IS_HERETIC_MONSTER(human_user))
```

File: code/modules/antagonists/eldritch_cult/eldritch_effects.dm
Comment (park66665) [inline code/modules/antagonists/eldritch_cult/eldritch_effects.dm]:
```suggestion
	QDEL_IN(src, 2 MINUTES)
```

File: code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm
Comment (park66665) [inline code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm]: This `dna` list is not used. Its usage isn't implemented in tg either. So you have a choice. You implement it, or you don't port the remnants of dna check.

File: code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm
Comment (park66665) [inline code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm:273]: Remind you, this change will conflict with #3451

Comment (eeSPee) [reply code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm:273]: Yes, #3451 should be implemented after this one.

This shouldn't be a problem.

File: code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm
Comment (eeSPee) [reply code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm]: I think this is a leftover of some sorts.

**Comment by utilbird** [comment]
Ah right, should probably merge this before we merge any other heretic PRs

**Comment by heepox** [comment]
i wonder if this will have to be atomized for anything to be merged...

Combat Tweak
- balance: Ash grasp blinds instead of knockback
- Heretics marks area work differently; marks are applied by grasp, and triggered with the blade

Tweaks to Gamemode in general
- tweak: Rust spread datum is revised/optimized
- tweak: Tears in reality no longer inflict brain damage when analyzed, instead just give a bad moodlet
- tweak: Revised reality tears generation
- tweak: Tears in reality appear after a time, and slowly disappear after 15 minutes
- balance: Heretic effects (runes, tears) are invisible to silicons
- spellcheck: Names and descriptions have been revised, thanks to @EOBGames

Heart Changes
- Heart now offers a pick between 3 targets
- Doubled the charges sacrifices offer

**Review by utilbird — DISMISSED** [review]

**Comment by eeSPee** [comment]
> i wonder if this will have to be atomized for anything to be merged...
> 
> Combat Tweak
> 
> * balance: Ash grasp blinds instead of knockback
> * Heretics marks area work differently; marks are applied by grasp, and triggered with the blade
> 
> Tweaks to Gamemode in general
> 
> * tweak: Rust spread datum is revised/optimized
> * tweak: Tears in reality no longer inflict brain damage when analyzed, instead just give a bad moodlet
> * tweak: Revised reality tears generation
> * tweak: Tears in reality appear after a time, and slowly disappear after 15 minutes
> * balance: Heretic effects (runes, tears) are invisible to silicons
> * spellcheck: Names and descriptions have been revised, thanks to @EOBGames
> 
> Heart Changes
> 
> * Heart now offers a pick between 3 targets
> * Doubled the charges sacrifices offer

It should be fine, and I don't see a reason why. All these changes are on TG.

File: code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm
Comment (park66665) [inline code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm]:
```suggestion
			for(var/i in 1 to 3)
```

Do you want to pick 4 targets?

**Review by park66665 — DISMISSED** [review]
Okay, the delay is slowly driving me mad. I'll put this into a vote in order to stimulate discussion with AUTH48: I *will* merge this PR in 48 hours unless another maintainer objects. The author is free to make minuscule last minute changes.

**Review by park66665 — APPROVED** [review]
The time has come, and the wiki is updated. This way to merging PRs isn't very ideal, but large PRs also should be merged some day, and we have to make that some day happen even if the stars fail to align.

All clear. This PR is good to go.

**Comment by eeSPee** [comment]
Thanks

**Pull request merged**
